src/tool.cpp

```diff
@@ -320,6 +320,9 @@ HitParams getHitParams(const ItemGroupList &armor_groups,
 		result_wear = calculateResultWear(tp->punch_attack_uses, initial_wear);
 		result_wear *= punch_interval_multiplier;
 	}
+	// Keep damage in sane bounds for simplicity
+	damage = MYMIN(damage, 65535);
+	damage = MYMAX(damage, -65535);
```

Comment:
"sane" bounds here would be s16 IMO
also you can use `rangelim(damage, S16_MIN, S16_MAX)` here

Reply:
Well then use (+/-)`U16_MAX`

Reply:
Bounds should be s16, at least because
1) on_punchplayer accepts s16
2) `(s32)getHP() - (s32)hitparams.hp` could overflow otherwise

Reply:
>     2. `(s32)getHP() - (s32)hitparams.hp` could overflow otherwise

nvm it couldn't because m_hp is s16. Fix on_punchplayer then and it should work.